### DISCUSSION TITLE
Tags for multi tenancy

### DISCUSSION THREAD

**Discussion by lepazca** [netbox-community/netbox#10382] [Q&A]
I'm trying to use tags for accomplish multi tenancy. The only thing, that I'm aware of, that prevent that a given user can see any other object except his is the possibility to list tagged objects. I maintained the possibility to view the created tags so they can use it during the creation of a new object.

Maybe this is not the right way to do this.

**Comment by cyberndj** [comment]
Howdy,
Might I suggest the option of using locations for your multi-tenancy? You can assign multiple locations to a site and also assign locations to other locations. You then could create your different tenants in the tenant module then assign them to each one of the locations.  An example setup would be:

Site: 1 Infinite Loop, Palo Alto CA
Locations: 
1 Infinite Loop (Assign to the Site)
- Floor 1
-  - Suite 101, Tenant: Customer 1
-  - Suite 102, Tenant: Customer 2
-  - MPOE
-  - 1st Floor MDF
- Floor 2
-  - Suite 201, Tenant: Customer 3
-  - Suite 202, Tenant: Customer 3
-  - 2nd Floor IDF

**Comment by lepazca** [comment]
Hi, the problem with this approach is that I pretend to use almost all Netbox's functionality in a multi-tenancy scenario. If I understand you correctly this will be limited only to sites.

**Comment by cyberndj** [comment]
I might have read your OP wrong. To Sum up, your just trying to do user level permissions so others can see their own tenancy?
If so, setting up a user group in the django admin page for each tenant based on a tag or tenant with all the restrictions and setting up the constraint section to limit access might be the way to go.
See the doc: https://docs.netbox.dev/en/stable/administration/permissions/ 

The tenant field can be applied to a lot more than sites/locations. After 10 min of looking, I  saw it couldn't be associated with contacts (although there could be other types too).
